pkg/kn/traffic/compute.go

```diff
@@ -197,53 +197,55 @@ func errorOverWritingTag(tag string) error {
 		"add flag '--untag %s' in command to untag it", tag)
 }
 
-func errorRepeatingLatestRevision(forFlag string) error {
+func errorRepeatingRevision(forFlag string, name string) error {
 	return fmt.Errorf("repetition of identifier %s "+
-		"is not allowed, use only once with %s flag", latestRevisionRef, forFlag)
+		"is not allowed, use only once with %s flag", name, forFlag)
 }
 
 // verifies if user has repeated @latest field in --tag or --traffic flags
 // verifyInputSanity checks:
 // - if user has repeated @latest field in --tag or --traffic flags
 // - if provided traffic portion are integers
 func verifyInputSanity(trafficFlags *flags.Traffic) error {
-	var latestRevisionTag = false
-	var latestRevisionTraffic = false
 	var sum = 0
 
-	for _, each := range trafficFlags.RevisionsTags {
+	// To heck if there are duplicated @latest in --tag flags
+	revisionMap := make(map[string]int)
+	for i, each := range trafficFlags.RevisionsTags {
 		revision, _, err := splitByEqualSign(each)
 		if err != nil {
 			return err
 		}
 
-		if latestRevisionTag && revision == latestRevisionRef {
-			return errorRepeatingLatestRevision("--tag")
-
-		}
-
-		if revision == latestRevisionRef {
-			latestRevisionTag = true
```

Comment:
We only disallow `@latest` in `--tag` flag, so we don't need to check for other revisions and create a map, only one bool variable should be good.

Reply:
Got it, thanks, will update it!